src/components/Fieldset/Fieldset.tsx

```diff
@@ -1,70 +1,89 @@
    */
-  legend?: string;
+  children: ReactNode;
   /**
-   * Slot for node to appear directly after legend text. Typically used to include a Toolip
+   * Additional classnames passed in for styling.
    */
-  legendAfter?: ReactNode;
+  className?: string;
+};
+
+type ItemsProps = {
   /**
-   * String for the optional label. By default it is '(optional)'
+   * The content of the control elements in the fieldset.
    */
-  optionalLabel?: string;
+  children: ReactNode;
   /**
-   * Required is passed down to the legend to display "optional" or not
+   * Type of element the immediate wrapper around the contents should be.
+   * @default 'div'
    */
-  required?: boolean;
+  as?: ElementType;
   /**
-   * String for the required label to add additional information if needed.
+   * Additional classnames passed in for styling.
    */
-  requiredLabel?: string;
-}
+  className?: string;
+};
 
 /**
- * Primary UI component for user interaction
+ * Helper sub-component for styling the title at the top of the component.
  */
-export const Fieldset = ({
-  className,
-  legend,
-  legendAfter,
-  hideLegend,
-  required,
+const Legend: React.FC<LegendProps> = ({ children, className }) => {
+  const componentClassName = clsx(className, styles['legend']);
+  return <legend className={componentClassName}>{children}</legend>;
+};
+
+/**
+ * Helper sub-component for styling the control elements in the component.
+ */
+const Items: React.FC<ItemsProps> = ({
   children,
-  optionalLabel,
-  requiredLabel,
-  ...other
-}: Props) => {
-  const componentClassName = clsx(styles['fieldset'], className, {});
-  return (
-    <fieldset className={componentClassName} {...other}>
-      <Legend
-        className={styles['fieldset__legend']}
-        text={legend}
-        required={required}
-        optionalLabel={optionalLabel}
-        requiredLabel={requiredLabel}
-        hideLegend={hideLegend}
-        legendAfter={legendAfter}
-      />
-      {children}
-    </fieldset>
-  );
+  as: Component = 'div',
+  className,
+}) => {
+  const componentClassName = clsx(className, styles['fieldset__items']);
+  return <Component className={componentClassName}>{children}</Component>;
 };
+
+/**
+ * ```ts
+ * import {Fieldset} from "@chanzuckerberg/eds-components";
+ * ```
+ *
+ * A container for a fieldset including a legend and one or more form inputs.
+ *
+ * Uses `fieldset` and `legend` elements by default but can be customized.
+ *
+ * If you're not using a `fieldset` element, please consider whether you should be
+ * using a `ul` instead.
+ */
+export function Fieldset({ children, className }: FieldsetProps) {
+  const componentClassName = clsx(className, styles['fieldset']);
+  return <fieldset className={componentClassName}>{children}</fieldset>;
+}
+
+Fieldset.Legend = Legend;
+Fieldset.Items = Items;
+
+Fieldset.displayName = 'Fieldset';
+Fieldset.Legend.displayName = 'Fieldset.Legend';
+Fieldset.Items.displayName = 'Fieldset.Items';
```

Comment:
nit: do we need these since we aren't forwarding refs?

Reply:
Probably not, will remove. These will also change with the components being extracted into own files